src/App.tsx

```diff
@@ -1,61 +1,149 @@
-import React from 'react';
+import React, { useState, useEffect, useCallback } from 'react';
+import cn from 'classnames';
 import 'bulma/bulma.sass';
 import '@fortawesome/fontawesome-free/css/all.css';
 import './App.scss';
 
-import classNames from 'classnames';
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { User } from './types/User';
+import { getUsers } from './api/users';
+import { Post } from './types/Post';
+import { getPosts } from './api/posts';
+import { ErrorNotification } from './components/ErrorNotification';
+import { Errors } from './types/Errors';
 
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isPostsLoaded, setIsPostsLoaded] = useState(false);
+  const [isPostsLoadingError, setIsPostsLoadingError] = useState(false);
+  const [selectedPostId, setSelectedPostId] = useState(0);
+  const [isNewCommentFormOpened, setIsNewCommentFormOpened] = useState(false);
+
+  useEffect(() => {
+    getUsers()
+      .then(setUsers);
+  }, []);
+
+  const selectUserId = useCallback((userId: number) => {
+    setSelectedUserId(userId);
+    setIsLoading(true);
+  }, []);
+
+  const selectPostId = useCallback((postId: number) => {
+    setSelectedPostId(prev => (prev !== postId ? postId : 0));
+    setIsNewCommentFormOpened(false);
+  }, []);
+
+  const resetSelectedPostId = useCallback((postId: number) => {
+    setSelectedPostId(postId);
+  }, []);
+
+  const openNewCommentForm = useCallback(() => {
+    setIsNewCommentFormOpened(true);
+  }, []);
+
+  useEffect(() => {
+    if (isLoading) {
+      getPosts(selectedUserId)
+        .then(loadedPosts => {
+          setPosts(loadedPosts);
+          setIsPostsLoaded(true);
+          setIsPostsLoadingError(false);
+        })
+        .catch(() => {
+          setIsPostsLoadingError(true);
+        })
+        .finally(() => setIsLoading(false));
+    }
+  }, [selectedUserId]);
+
+  const isLoadedPostsExist = !isLoading
+    && !isPostsLoadingError
+    && isPostsLoaded
+    && posts.length > 0;
+
+  const isLoadedPostsNotExist = !isLoading
+    && !isPostsLoadingError
+    && isPostsLoaded
+    && !posts.length;
```

Comment:
it's not critical, but do you really need two of these variables?

Reply:
the same here